library/oid.c

```diff
@@ -861,4 +895,168 @@ int mbedtls_oid_get_numeric_string(char *buf, size_t size,
     return (int) (size - n);
 }
 
+static int oid_parse_number(const char **p, const char *bound)
+{
+    int ret = MBEDTLS_ERR_ASN1_INVALID_DATA;
+    int num = 0;
```

Comment:
why `int` rather than `unsigned` (and therefore also `UINT_MAX` at line 904)?

Reply:
To answer my own question: because you're using in-band error reporting on the single `int` result of the function. You could change this function to take an `unsigned *` (first parameter) to write the number to, and then return a standard `0` for success